File: frontend/lib/types/userStatus.ts
// User Status Types
export enum UserStatus {
  ONLINE = 'ONLINE',
  AWAY = 'AWAY',
  OFFLINE = 'OFFLINE'
}

export interface UserStatusDto {
  userId: number;
  username: string;
  displayName: string;
  avatarUrl?: string;
  status: UserStatus;
  lastSeenAt?: string;
  lastActivityAt?: string;
}

export interface UserWithStatus {
  id: number;
  username: string;
  displayName: string;
  email: string;
  avatarUrl?: string;
  isOnline: boolean;
  status: UserStatus;
  lastSeenAt?: string;
  lastActivityAt?: string;
}

File: backend/src/main/java/com/app/pverse/entity/UserStatus.java
package com.app.pverse.entity;

/**
 * User Status Enum
 * Định nghĩa các trạng thái của user
 */
public enum UserStatus {
    ONLINE,     // Đang hoạt động
    AWAY,       // Vắng mặt (không tương tác > 5 phút)
    OFFLINE     // Không trực tuyến
}

